VirtoCommerce.Platform.Core/Common/IQueryableExtensions.cs

```diff
@@ -68,40 +67,42 @@ public static IOrderedQueryable<T> ThenBySortInfos<T>(this IOrderedQueryable<T>
         }
 
         public static IOrderedQueryable<T> ApplyOrder<T>(IQueryable<T> source, string property, string methodName)
-		{
-			if (property == null)
-				throw new ArgumentNullException("property");
+        {
+            if (property == null)
+                throw new ArgumentNullException(nameof(property));
+
+            var props = property.Split('.');
+            var registeredType = AbstractTypeFactory<T>.AllTypeInfos.Select(x => x.Type).FirstOrDefault();
```

Comment:
Better to use AbstractTypeFactory<T>.TryCreateInstance(), as it consider inheritance.

Reply:
fix